src/Conversion/ONNXToTOSA/NN/MaxPoolSingleOut.cpp

```diff
@@ -132,10 +132,13 @@ class ONNXMaxPoolSingleOutOpLoweringToTOSA : public ConversionPattern {
     IndexExpr::getLiteral(shapeHelper.pads, pads);
 
     // reorder padding values
-    mlir::ArrayAttr newPads = rewriter.getI64ArrayAttr({pads[0],
+    auto newPads = rewriter.getDenseI64ArrayAttr({pads[0],
         pads[2] + ceilConstants[0], pads[1], pads[3] + ceilConstants[1]});
 
-    mlir::ArrayAttr strides = rewriter.getI64ArrayAttr(shapeHelper.strides);
+    auto strides = rewriter.getDenseI64ArrayAttr(shapeHelper.strides);
+    auto kernelShape = rewriter.getDenseI64ArrayAttr(llvm::to_vector(
```

Comment:
Since kernelShape is already an I64ArrayAttr I believe you can use the helper here:
`auto kernelShape = rewriter.getDenseI64ArrayAttr(extractFromI64ArrayAttr(adaptor.getKernelShapeAttr()));`